### PR TITLE
Do not filter id matching userId on non-user-share shares

### PR DESCRIPTION
**Pull request by skjnldsv**
Regression from #15952

Before
Kazam_screenshot_00018
___
After
Kazam_screenshot_00017

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -791,7 +791,8 @@ private function getFormattedShares(
 				continue;
 			}
 
-			if (in_array($share->getId(), $known) || $share->getSharedWith() === $this->currentUser) {
+			if (in_array($share->getId(), $known)
```
Comment (ChristophWurst) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:794]: mind adding parenthesis so it's either clearly (A||B)&&C or A||(B&&C)?

Comment (skjnldsv) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:794]: Done!

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Not build with production

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
> Lint / php-cs check (pull_request) Failing

**Comment by skjnldsv** [comment]
> > Lint / php-cs check (pull_request) Failing

fixed

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Pull request merged**